examples/flow-counter/types.js

```diff
@@ -0,0 +1,18 @@
+/* @flow */
+
+import type { ThunkMiddlewareDispatchable } from 'redux-thunk';
+import type { ReduxAction } from 'redux';
+
+export type CounterState = number;
+
+export type AppState = {
+  counter: CounterState
+}
+
+// Need to explicitly put this here due to a bug in Flow: https://github.com/facebook/flow/issues/582
+export type Action =
+  { type: '@@redux/INIT'} |
```

Comment:
As noted here, I still need to specify this internal type because of the bug noted above that makes Flow not recognize nested union types correctly.